app/src/main/kotlin/de/digitalService/useID/util/AbTestManager.kt

```diff
@@ -0,0 +1,80 @@
+package de.digitalService.useID.util
+
+import androidx.compose.runtime.State
+import androidx.compose.runtime.mutableStateOf
+import de.digitalService.useID.analytics.IssueTrackerManagerType
+import de.digitalService.useID.analytics.TrackerManagerType
+import de.digitalService.useID.hilt.ConfigModule
+import io.getunleash.UnleashClient
+import io.getunleash.UnleashConfig
+import io.getunleash.UnleashContext
+import io.getunleash.polling.PollingModes
+import kotlinx.coroutines.suspendCancellableCoroutine
+import java.util.*
+import javax.inject.Inject
+import javax.inject.Named
+import javax.inject.Singleton
+import kotlin.coroutines.resume
+
+@Singleton
+class AbTestManager @Inject constructor(
+    @Named(ConfigModule.UNLEASH_API_URL) private val url: String,
+    @Named(ConfigModule.UNLEASH_API_KEY) private val apiKey: String,
+    private val trackerManager: TrackerManagerType,
+    private val issueTrackerManager: IssueTrackerManagerType
+) {
+
+    private lateinit var unleashClient: UnleashClient
+
+    private val _isSetupIntroTestVariant = mutableStateOf(false)
+    val isSetupIntroTestVariant: State<Boolean> = _isSetupIntroTestVariant
+
+    suspend fun initialise() = suspendCancellableCoroutine { continuation ->
+        val config = UnleashConfig.newBuilder()
+            .proxyUrl(url)
+            .clientKey(apiKey)
+            .pollingMode(PollingModes.autoPoll(Int.MAX_VALUE.toLong()))
+            .build()
+
+        val supportedAbTests = AbTest.values().joinToString(",") { it.testName }
+        val context = UnleashContext.newBuilder()
+            .appName("useid-android")
+            .sessionId(UUID.randomUUID().toString())
```

Comment:
Oh, I haven’t done this and I see a weird issue when variation sticks even though there is no consistent value being sent from our side. Session id being different every time might actually help, because it feels like unleash has some caching for when it’s not present 💡